From bbb05d976d16305c54c9a6bf45a1cb08d0536bf9 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Fri, 10 Jun 2022 13:37:21 +0300
Subject: [PATCH] [SW-70527] gaudi2: ras: add missing pattern checks

Two instances of pattern check were missing in
gaudi2_hmmu_xbar_edge_mc_ring_pattern().
Added those checks in this patch.

Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
Change-Id: I9ff5841ac7a015a7cf0b9eb4d55605fa64773db6
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index da695c9..a6b3d35 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -718,11 +718,13 @@
 
 		expected_ones = 0b00000000000000000000000000000000;
 		expected_zeros = 0b00000000100100100100100100100100;
+		gaudi2_sms_read_and_check_pattern(expected_ones, expected_zeros);
 
 		regmap_writel(GAUDI2_RING_PATTERN_VAL7, mmPSOC_DFT_APB_WORD1);
 
 		expected_ones = 0b00000000000000000000000000000000;
 		expected_zeros = 0b00000000000000000000000010010000;
+		gaudi2_sms_read_and_check_pattern(expected_ones, expected_zeros);
 	}
 }
 
